### PR TITLE
TST / Workflows: Get slack notifications for docker image build

### PR DESCRIPTION
**Pull request by younesbelkada**
# What does this PR do?

As per title, as similarly as https://github.com/huggingface/peft/pull/1723 - I created a new slack channel to receive notifications on the docker image build status for all our docker images 

We do that in PEFT / diffusers (https://github.com/huggingface/diffusers/pull/7938)

Related to: https://github.com/huggingface/transformers/pull/30890

cc @ydshieh

### CODE REVIEW

File: .github/workflows/build-docker-images.yml [hunk 1]
```diff
@@ -57,6 +57,15 @@ jobs:
           push: true
           tags: huggingface/transformers-all-latest-gpu-push-ci
 
+      - name: Post to Slack
+        if: always()
+        uses: huggingface/hf-workflows/.github/actions/post-slack@main
+        with:
+          slack_channel: "C073ZTN10A1"
```
Comment (younesbelkada) [inline .github/workflows/build-docker-images.yml:64]: Change to correct env name in the future and use a secret

**Review by SunMarc — APPROVED** [review]
Good idea ! Thx for adding this !

File: .github/workflows/build-docker-images.yml (hunk 1 above)
Comment (younesbelkada) [inline .github/workflows/build-docker-images.yml:64]:
```suggestion
          slack_channel: ${{ secrets.CI_DOCKER_CHANNEL }}
```

File: .github/workflows/build-docker-images.yml [hunk 2]
```diff
@@ -93,6 +102,15 @@ jobs:
           push: true
           tags: huggingface/transformers-pytorch-deepspeed-latest-gpu${{ inputs.image_postfix }}
 
+      - name: Post to Slack
+        if: always()
+        uses: huggingface/hf-workflows/.github/actions/post-slack@main
+        with:
+          slack_channel: "C073ZTN10A1"
```
Comment (younesbelkada) [inline .github/workflows/build-docker-images.yml:109]:
```suggestion
          slack_channel: ${{ secrets.CI_DOCKER_CHANNEL }}
```

File: .github/workflows/build-docker-images.yml [hunk 3]
```diff
@@ -134,6 +152,15 @@ jobs:
           push: true
           tags: huggingface/transformers-pytorch-deepspeed-latest-gpu-push-ci
 
+      - name: Post to Slack
+        if: always()
+        uses: huggingface/hf-workflows/.github/actions/post-slack@main
+        with:
+          slack_channel: "C073ZTN10A1"
```
Comment (younesbelkada) [inline .github/workflows/build-docker-images.yml:159]:
```suggestion
          slack_channel: ${{ secrets.CI_DOCKER_CHANNEL }}
```

File: .github/workflows/build-docker-images.yml [hunk 4]
```diff
@@ -160,6 +187,15 @@ jobs:
           push: true
           tags: huggingface/transformers-doc-builder
 
+      - name: Post to Slack
+        if: always()
+        uses: huggingface/hf-workflows/.github/actions/post-slack@main
+        with:
+          slack_channel: "C073ZTN10A1"
```
Comment (younesbelkada) [inline .github/workflows/build-docker-images.yml:194]:
```suggestion
          slack_channel: ${{ secrets.CI_DOCKER_CHANNEL }}
```

File: .github/workflows/build-docker-images.yml [hunk 5]
```diff
@@ -198,6 +234,15 @@ jobs:
           push: true
           tags: huggingface/transformers-pytorch-gpu
 
+      - name: Post to Slack
+        if: always()
+        uses: huggingface/hf-workflows/.github/actions/post-slack@main
+        with:
+          slack_channel: "C073ZTN10A1"
```
Comment (younesbelkada) [inline .github/workflows/build-docker-images.yml:241]:
```suggestion
          slack_channel: ${{ secrets.CI_DOCKER_CHANNEL }}
```

File: .github/workflows/build-docker-images.yml [hunk 6]
```diff
@@ -237,6 +282,15 @@ jobs:
           push: true
           tags: huggingface/transformers-pytorch-amd-gpu-push-ci
 
+      - name: Post to Slack
+        if: always()
+        uses: huggingface/hf-workflows/.github/actions/post-slack@main
+        with:
+          slack_channel: "C073ZTN10A1"
```
Comment (younesbelkada) [inline .github/workflows/build-docker-images.yml:289]:
```suggestion
          slack_channel: ${{ secrets.CI_DOCKER_CHANNEL }}
```

File: .github/workflows/build-docker-images.yml [hunk 7]
```diff
@@ -265,6 +319,15 @@ jobs:
           push: true
           tags: huggingface/transformers-tensorflow-gpu
 
+      - name: Post to Slack
+        if: always()
+        uses: huggingface/hf-workflows/.github/actions/post-slack@main
+        with:
+          slack_channel: "C073ZTN10A1"
```
Comment (younesbelkada) [inline .github/workflows/build-docker-images.yml:326]:
```suggestion
          slack_channel: ${{ secrets.CI_DOCKER_CHANNEL }}
```

File: .github/workflows/build-docker-images.yml [hunk 8]
```diff
@@ -304,6 +367,15 @@ jobs:
           push: true
           tags: huggingface/transformers-pytorch-deepspeed-amd-gpu-push-ci
 
+      - name: Post to Slack
+        if: always()
+        uses: huggingface/hf-workflows/.github/actions/post-slack@main
+        with:
+          slack_channel: "C073ZTN10A1"
```
Comment (younesbelkada) [inline .github/workflows/build-docker-images.yml:374]:
```suggestion
          slack_channel: ${{ secrets.CI_DOCKER_CHANNEL }}
```

File: .github/workflows/build-docker-images.yml [hunk 9]
```diff
@@ -330,4 +402,13 @@ jobs:
...
           push: true
-          tags: huggingface/transformers-quantization-latest-gpu${{ inputs.image_postfix }}
\ No newline at end of file
+          tags: huggingface/transformers-quantization-latest-gpu${{ inputs.image_postfix }}
+
+      - name: Post to Slack
+        if: always()
+        uses: huggingface/hf-workflows/.github/actions/post-slack@main
+        with:
+          slack_channel: "C073ZTN10A1"
```
Comment (younesbelkada) [inline .github/workflows/build-docker-images.yml:411]:
```suggestion
          slack_channel: ${{ secrets.CI_DOCKER_CHANNEL }}
```

File: .github/workflows/build-docker-images.yml [hunk 10]
```diff
@@ -57,6 +57,15 @@ jobs:
           push: true
           tags: huggingface/transformers-all-latest-gpu-push-ci
 
+      - name: Post to Slack
+        if: always()
+        uses: huggingface/hf-workflows/.github/actions/post-slack@main
+        with:
+          slack_channel: ${{ secrets.CI_DOCKER_CHANNEL }}
```
Comment (younesbelkada) [inline .github/workflows/build-docker-images.yml:64]:
```suggestion
          slack_channel: ${{ secrets.CI_SLACK_CHANNEL_DOCKER }}
```

File: .github/workflows/build-docker-images.yml [hunk 11]
```diff
@@ -93,6 +102,15 @@ jobs:
           push: true
           tags: huggingface/transformers-pytorch-deepspeed-latest-gpu${{ inputs.image_postfix }}
 
+      - name: Post to Slack
+        if: always()
+        uses: huggingface/hf-workflows/.github/actions/post-slack@main
+        with:
+          slack_channel: ${{ secrets.CI_DOCKER_CHANNEL }}
```
Comment (younesbelkada) [inline .github/workflows/build-docker-images.yml:109]:
```suggestion
          slack_channel: ${{ secrets.CI_SLACK_CHANNEL_DOCKER}}
```

File: .github/workflows/build-docker-images.yml [hunk 12]
```diff
@@ -134,6 +152,15 @@ jobs:
           push: true
           tags: huggingface/transformers-pytorch-deepspeed-latest-gpu-push-ci
 
+      - name: Post to Slack
+        if: always()
+        uses: huggingface/hf-workflows/.github/actions/post-slack@main
+        with:
+          slack_channel: ${{ secrets.CI_DOCKER_CHANNEL }}
```
Comment (younesbelkada) [inline .github/workflows/build-docker-images.yml:159]:
```suggestion
          slack_channel: ${{ secrets.CI_SLACK_CHANNEL_DOCKER }}
```

File: .github/workflows/build-docker-images.yml [hunk 13]
```diff
@@ -160,6 +187,15 @@ jobs:
           push: true
           tags: huggingface/transformers-doc-builder
 
+      - name: Post to Slack
+        if: always()
+        uses: huggingface/hf-workflows/.github/actions/post-slack@main
+        with:
+          slack_channel: ${{ secrets.CI_DOCKER_CHANNEL }}
```
Comment (younesbelkada) [inline .github/workflows/build-docker-images.yml:194]:
```suggestion
          slack_channel: ${{ secrets.CI_SLACK_CHANNEL_DOCKER }}
```

File: .github/workflows/build-docker-images.yml [hunk 14]
```diff
@@ -198,6 +234,15 @@ jobs:
           push: true
           tags: huggingface/transformers-pytorch-gpu
 
+      - name: Post to Slack
+        if: always()
+        uses: huggingface/hf-workflows/.github/actions/post-slack@main
+        with:
+          slack_channel: ${{ secrets.CI_DOCKER_CHANNEL }}
```
Comment (younesbelkada) [inline .github/workflows/build-docker-images.yml:241]:
```suggestion
          slack_channel: ${{ secrets.CI_SLACK_CHANNEL_DOCKER }}
```

File: .github/workflows/build-docker-images.yml [hunk 15]
```diff
@@ -237,6 +282,15 @@ jobs:
           push: true
           tags: huggingface/transformers-pytorch-amd-gpu-push-ci
 
+      - name: Post to Slack
+        if: always()
+        uses: huggingface/hf-workflows/.github/actions/post-slack@main
+        with:
+          slack_channel: ${{ secrets.CI_DOCKER_CHANNEL }}
```
Comment (younesbelkada) [inline .github/workflows/build-docker-images.yml:289]:
```suggestion
          slack_channel: ${{ secrets.CI_SLACK_CHANNEL_DOCKER }}
```

File: .github/workflows/build-docker-images.yml [hunk 16]
```diff
@@ -265,6 +319,15 @@ jobs:
           push: true
           tags: huggingface/transformers-tensorflow-gpu
 
+      - name: Post to Slack
+        if: always()
+        uses: huggingface/hf-workflows/.github/actions/post-slack@main
+        with:
+          slack_channel: ${{ secrets.CI_DOCKER_CHANNEL }}
```
Comment (younesbelkada) [inline .github/workflows/build-docker-images.yml:326]:
```suggestion
          slack_channel: ${{ secrets.CI_SLACK_CHANNEL_DOCKER }}
```

File: .github/workflows/build-docker-images.yml [hunk 17]
```diff
@@ -304,6 +367,15 @@ jobs:
           push: true
           tags: huggingface/transformers-pytorch-deepspeed-amd-gpu-push-ci
 
+      - name: Post to Slack
+        if: always()
+        uses: huggingface/hf-workflows/.github/actions/post-slack@main
+        with:
+          slack_channel: ${{ secrets.CI_DOCKER_CHANNEL }}
```
Comment (younesbelkada) [inline .github/workflows/build-docker-images.yml:374]:
```suggestion
          slack_channel: ${{ secrets.CI_SLACK_CHANNEL_DOCKER }}
```

**Review by ydshieh — APPROVED** [review]
Very nice.

It sends many separate reports for a workflow run however. Ideally it might be a good idea to send all of these in on go (in a final job), but it will require some changes in `hf-workflows`. So I wouldn't insist such change in this PR.

**Comment by younesbelkada** [comment]
Thanks so much for the review @ydshieh !

**Pull request merged**